llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelCacheImpl.java

```diff
@@ -305,10 +305,10 @@ private boolean lockBuffer(LlapDataBuffer buffer, boolean doNotifyPolicy) {
         long offset = ranges[i].getOffset() + baseOffset;
         assert buffer.declaredCachedLength == LlapDataBuffer.UNKNOWN_CACHED_LENGTH;
         buffer.declaredCachedLength = ranges[i].getLength();
-        buffer.setTag(tag);
         while (true) { // Overwhelmingly executes once, or maybe twice (replacing stale value).
           LlapDataBuffer oldVal = subCache.getCache().putIfAbsent(offset, buffer);
           if (oldVal == null) {
+            buffer.setFileCache(subCache);
```

Comment:
This sets the reference from buffer to filecache instance alright.
But as far as I see this only happens when LowLevelCacheImpl is used, aka caching Orc files, how does this reference setting happen with the SerdeLowLevelCache (text based files)?

Reply:
Thanks, it's fixed now.